pkg/credentials/provider.go

```diff
@@ -62,7 +65,7 @@ func (sp *Provider) Retrieve() (credentials.Value, error) {
 
 	out, err := sp.assumeRole(ctx, token)
 	if err != nil {
-		return credentials.Value{}, err
+		return credentials.Value{}, errors.Wrap(err, "failed to assume role")
 	}
```

Comment:
As mentioned previously, Go doesn't seem to like stack traces out of the box. errors.Wrap() preserves them for easy printing.